src/settings_gui.cpp

```diff
@@ -140,6 +142,52 @@ void ShowBaseSetTextfileWindow(TextfileType file_type, const TBaseSet* baseset,
 	new BaseSetTextfileWindow<TBaseSet>(file_type, baseset, content_type);
 }
 
+std::vector<int> refresh_rates = { 30, 60, 75, 90, 100, 120, 144, 240 };
+
+// We want 60Hz as the default for this pointer, though it gets overriden with whatever is in the config.
+int *current_refresh_rate = &refresh_rates[1];
+
+std::string hertz = "Hz";
+std::string current_refresh_rate_str = std::to_string(*current_refresh_rate) + hertz;
+
+// Add the refresh rate from the config and the refresh rates from all the monitors to
+// our list of refresh rates shown in the GUI.
+void addRefreshRatesAndSelect() {
+	// std::unordered_set so we can efficiently add all monitor refresh rates to this
+	// set without duplicates, as it has near constant lookup times.
+	std::unordered_set<int> tempRates(refresh_rates.begin(), refresh_rates.end());
```

Comment:
You always expend on the list, also from earlier runs. Would it be an idea to make `REFRESH_RATE` a constant, and fill `refresh_rate` based on that, and anything detected? That would mean that if the system changes during runtime, old values are not left behind. Also means you can do away with this temporary variable, and clear `refresh_rates` at the beginning of this function. Might clean up the code a bit? Dunno, up to you :)

Reply:
I really don't see the benefits and the temporary variable will still be needed, as when detecting refresh rates duplicates can arise. Also extending on the previous list allows older refresh rates to technically be used, I don't see a reason to not show these.